File: src/main/java/jb/com/CouponSystemPhase2/services/CompanyService.java
package jb.com.CouponSystemPhase2.services;

import jb.com.CouponSystemPhase2.beans.Category;
import jb.com.CouponSystemPhase2.beans.Company;
import jb.com.CouponSystemPhase2.beans.Coupon;
import jb.com.CouponSystemPhase2.exceptions.CouponSystemException;

import java.util.List;
import java.util.Optional;

public interface CompanyService {
   void addCoupon(int companyId,Coupon coupon) throws CouponSystemException;
   void updateCoupon(int companyId, int couponId, Coupon coupon) throws CouponSystemException;
   void deleteCoupon(int companyId, int couponId) throws CouponSystemException;
   List<Coupon> getAllCouponsByCategory( int companyId, Category category) throws CouponSystemException;
   List<Coupon> getAllCouponsByMaxPrice(int companyId, double maxPrice) throws CouponSystemException;
   Optional<Company> getCompanyInfo(int companyId) throws CouponSystemException;
   Optional<Coupon> getCoupon(int companyId, int couponId) throws CouponSystemException;
   List<Coupon> getAllCoupons(int companyId);
}

File: couponology/src/Services/CompanyService.ts
import axios from "axios";
import { CouponModel } from "../Models/Coupon";
import { CompanyModel } from "../Models/Company";
import { LoginReq } from "../Components/AuthArea/Login/Login";


class CompanyService {
  public async getAllCoupons() {
    return (await axios.get<CouponModel[]>('http://localhost:8080/api/company/allCoupons')).data;
  }
  public async getAllCouponsByCategory(selectedCategory: string) {
    return (await axios.get<CouponModel[]>('http://localhost:8080/api/company/allCouponsByCategory?category=' + selectedCategory)).data;
  }
  public async getAllCouponsByMaxPrice(maxPrice: number) {
    return (await axios.get<CouponModel[]>('http://localhost:8080/api/company/allCouponsByMaxPrice?maxPrice=' + maxPrice)).data;
  }
  public async getInfo() {
    return (await axios.get<CompanyModel>('http://localhost:8080/api/company/companyInfo')).data;
  }
  public async getCoupon(id: number) {
    return (await axios.get<CouponModel>('http://localhost:8080/api/company/coupon/' + id)).data;
  }
  public async updateCoupon(coupon: CouponModel) {
    return await axios.put('http://localhost:8080/api/company/coupon/' + coupon.id, coupon)
  }
  public async deleteCoupon(id: number) {
    return (await axios.delete('http://localhost:8080/api/company/coupon/' + id)).data;
  }
  public async addCoupon(coupon: CouponModel) {
    return await axios.post('http://localhost:8080/api/company/coupons', coupon);
  }
  public async login(data: LoginReq) {
    return (await axios.post<{ token: string }>('http://localhost:8080/api/company/login', data));
  }
}



export default CompanyService;